pkg/api/base.go

```diff
@@ -36,3 +42,60 @@ func (this *BasePortal) GetParameters() (map[string][]string, error) {
 	}
 	return m, nil
 }
+
+type ActionSpec map[string]interface{}
+
+func (this *BasePortal) Action() {
```

Comment:
Can you explain why add this method?

Reply:
This is used for action router.
The replication enable, disable failover hvae the same url(e.g. http://localhost:50040/v1beta/ef305038-cd12-4f3b-90bd-0612f83e14ee/block/replications/$1/action ), so wo need to distinguish them by body keyword.

Some body examples will be like this:
```
Disable body:
{
    "disableReplication": {}
}'

Enable body:
{
    "enableReplication": {}
}'

Failover body:
{
    "failoverReplication": {
        "allowAttachedVolume": true,
        "secondaryBackendId": "vendor-id-1"
    }
}'
```

It's a very general function so it may be used in other feature such as volume extend.

Reply:
But for BaseModel, it's used for all resource models. If you think it can be ultilized by all resource, please add some introduction of this method in the code.

Reply:
That's fine.